### PR TITLE
[GLUTEN-5643] Fix the failure when the pre-project of GenerateExec falls back

### PR DESCRIPTION
**Pull request by marin-ma**
Currently, if GenerateExec needs a pre-project, only ProjectNode is allowed in the native substrait -> Velox PlanNode transformation. Query would fail if the pre-project of GenerateExec falls back due to native validation failure. We should consider `ValueStreamNode` as valid in case of pre-project fallback.

Introduce `VeloxDummyExpression` for test purpose. An expression wrapped by `VeloxDummyExpression` can mock the case when a certain expression is supported by Gluten, but it fails on native validation.

### CODE REVIEW

**Comment by marin-ma** [comment]
Run Gluten Clickhouse CI

**Comment by marin-ma** [comment]
@xumingming Could you help to review? Thanks!

**Comment by xumingming** [comment]
> @xumingming Could you help to review? Thanks!

Sure, will do the first round of review in the recent days.

**Comment by xumingming** [comment]
> Query would fail if the pre-project of GenerateExec falls back due to native validation failure. We should consider ValueStreamNode as valid in case of pre-project fallback.

@marin-ma I understand Project might fallback(some function not supported?), but I don't understand why we should consider ValueStreamNode as valid,  can you elaborate?

**Comment by marin-ma** [comment]
@xumingming `ValueStreamNode` represents the input iterator is not from the pipeline. With pre-project fallback, it represents the input is from RowToColumnar conversion.

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc [hunk 1]
```diff
@@ -752,9 +761,10 @@ core::PlanNodePtr SubstraitToVeloxPlanConverter::toVeloxPlan(const ::substrait::
       SubstraitParser::configSetInOptimization(generateRel.advanced_extension(), "injectedProject=");
 
   if (injectedProject) {
-    auto projNode = std::dynamic_pointer_cast<const core::ProjectNode>(childNode);
+    // Child should be either ProjectNode or ValueStreamNode in case of project fallback.
     VELOX_CHECK(
-        projNode != nullptr && projNode->names().size() > requiredChildOutput.size(),
+        std::dynamic_pointer_cast<const core::ProjectNode>(childNode) != nullptr ||
+            std::dynamic_pointer_cast<const ValueStreamNode>(childNode) != nullptr,
         "injectedProject is true, but the Project is missing or does not have the corresponding projection field")
```
Comment (xumingming) [inline cpp/velox/substrait/SubstraitToVeloxPlan.cc:768]: The error message need to be updated to reflect the code change.

**Review by zhztheplayer — APPROVED** [review]

**Pull request merged**